File: 228_summary_ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        if (nums.size()==0) return result;
        if (nums.size()==1) {result.push_back(to_string(nums[0]));return result;}
        int start=nums[0],end;
        for (int i=1;i<nums.size();i++ ) {
                if (nums[i]-nums[i-1]!=1) {
                    end=nums[i-1];
                    if (start==end) 
                    result.push_back(to_string(start));
                    else result.push_back(to_string(start)+ "->" + to_string(end));
                    start=nums[i];
                }
                
            }
            end=nums.back();
             if (start==end) 
                    result.push_back(to_string(start));
                    else result.push_back(to_string(start)+ "->" + to_string(end));
                    return result;
                
            }
};

File: algorithms/python3/228_summary_ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return []
        res=[]
        low,high=nums[0],nums[0]
        for i in range(1,len(nums)):
            if nums[i]-nums[i-1]>1:
                if low==high:
                    res.append(str(low))
                else:
                    res.append(str(low)+'->'+str(high))
                low=high=nums[i]
            else:
                high=nums[i]
        if low==high:
           res.append(str(low))
        else:
           res.append(str(low)+'->'+str(high))
        return res

a=Solution()

nums=[0,1,2,4,5,7]
print(a.summaryRanges(nums))


